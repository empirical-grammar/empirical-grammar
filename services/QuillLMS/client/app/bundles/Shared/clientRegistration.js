import 'lazysizes';
import 'lazysizes/plugins/parent-fit/ls.parent-fit';
import ReactOnRails from 'react-on-rails';
import './styles/styles.scss'

import {
  QuestionList,
  LinkListItem,
  QuestionListByConcept,
  Instructions,
  WarningDialogue,
  Prompt,
  ConceptExplanation,
  ArchivedButton,
  Card,
  CarouselAnimation,
  DataTable,
  ProgressBar,
  DropdownInput,
  Error,
  FlagDropdown,
  Input,
  Passthrough,
  ResumeOrBeginButton,
  SmartSpinner,
  Snackbar,
  defaultSnackbarTimeout,
  Spinner,
  TextEditor,
  TextArea,
  Tooltip,
  UploadOptimalResponses,
  Modal,
  ListItem,
  SortableList,
  AffectedResponse,
  PieChart,
  QuestionBar,
  ResponseSortFields,
  ResponseToggleFields,
  Feedback,
  Cue,
  CueExplanation,
  ThankYou,
  SentenceFragments,
  QuestionRow,
  MultipleChoice,
  Register,
  PlayTitleCard,
  TitleCard,
  hashToCollection,
  isValidRegex
} from './index'

ReactOnRails.register({
  QuestionList,
  LinkListItem,
  QuestionListByConcept,
  Instructions,
  WarningDialogue,
  Prompt,
  ConceptExplanation,
  ArchivedButton,
  Card,
  CarouselAnimation,
  DataTable,
  ProgressBar,
  DropdownInput,
  Error,
  FlagDropdown,
  Input,
  Passthrough,
  ResumeOrBeginButton,
  SmartSpinner,
  Snackbar,
  defaultSnackbarTimeout,
  Spinner,
  TextEditor,
  TextArea,
  Tooltip,
  UploadOptimalResponses,
  Modal,
  ListItem,
  SortableList,
  AffectedResponse,
  PieChart,
  QuestionBar,
  ResponseSortFields,
  ResponseToggleFields,
  Feedback,
  Cue,
  CueExplanation,
  ThankYou,
  SentenceFragments,
  QuestionRow,
  MultipleChoice,
  Register,
  PlayTitleCard,
  TitleCard,
  hashToCollection,
  isValidRegex
});
