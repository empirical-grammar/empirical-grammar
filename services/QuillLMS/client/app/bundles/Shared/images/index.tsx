export const bigCheckIcon = {
  src: `${process.env.CDN_URL}/images/icons/icons-check-big.svg`,
  alt: "White checkmark in green circle"
}

export const searchMapIcon = {
  src: `${process.env.CDN_URL}/images/icons/icons-map-search.svg`,
  alt: "Map with a magnifying glass icon"
}

export const clipboardCheckIcon = {
  src: `${process.env.CDN_URL}/images/icons/icons-clipboard-check.svg`,
  alt: "Clipboard with a checkmark icon"
}

export const tableCheckIcon = {
  src: `${process.env.CDN_URL}/images/icons/icons-table-check.svg`,
  alt: "Table with a checkmark icon"
}

export const accountViewIcon = {
  src: `${process.env.CDN_URL}/images/icons/icons-account-view.svg`,
  alt: "Avatar with an eye next to it icon"
}

export const giftIcon = {
  src: `${process.env.CDN_URL}/images/icons/icons-gift.svg`,
  alt: "Gift icon"
}

export const groupAccountIcon = {
  src: `${process.env.CDN_URL}/images/icons/icons-account-group.svg`,
  alt: "Three avatars icon"
}

export const googleClassroomIcon = {
  src: `${process.env.CDN_URL}/images/icons/icons-google-classroom.svg`,
  alt: "Google Classroom icon"
}
