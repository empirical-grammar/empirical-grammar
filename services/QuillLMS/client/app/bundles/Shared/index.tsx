export {
  QuestionList,
  LinkListItem,
  QuestionListByConcept
} from './components/questionList/index'

export {
  Instructions,
  Prompt
} from './components/fillInBlank/index'

export {
  ConceptExplanation,
} from './components/feedback/conceptExplanation'

export {
  ArchivedButton,
  Card,
  CarouselAnimation,
  DataTable,
  ProgressBar,
  DropdownInput,
  Error,
  FlagDropdown,
  Input,
  Passthrough,
  ResumeOrBeginButton,
  SmartSpinner,
  Snackbar,
  defaultSnackbarTimeout,
  Spinner,
  SortableList,
  TeacherPreviewMenu,
  TextEditor,
  TextArea,
  Tooltip,
  TwoThumbSlider,
  UploadOptimalResponses
} from './components/shared/index'

export {
  Modal
} from './components/modal/modal'

export {
  AffectedResponse,
  PieChart,
  QuestionBar,
  ResponseSortFields,
  ResponseToggleFields
} from './components/questions/index'

export {
  Feedback,
  Cue,
  CueExplanation,
  ThankYou,
  SentenceFragments
} from './components/renderForQuestions/index'

export {
  QuestionRow
} from './components/scoreAnalysis/index'

export {
  MultipleChoice,
  Register,
  PlayTitleCard
} from './components/studentLessons/index'

export {
  TitleCard
} from './components/titleCards/index'

export {
  hashToCollection,
  isValidRegex,
  momentFormatConstants,
  copyToClipboard
} from './libs/index'
