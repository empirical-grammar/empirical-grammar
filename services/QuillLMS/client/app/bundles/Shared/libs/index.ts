export { hashToCollection } from './hashToCollection'
export { isValidRegex } from './isValidRegex'
export { momentFormatConstants } from './momentFormatConstants'
export { copyToClipboard } from './copyToClipboard'
