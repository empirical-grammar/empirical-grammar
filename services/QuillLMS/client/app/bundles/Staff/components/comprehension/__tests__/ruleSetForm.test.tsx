import * as React from 'react';
import { shallow } from 'enzyme';
import RuleSetForm from '../configureRegex/ruleSetForm';
import RegexSection from '../configureRegex/regexSection';
import { Input, TextEditor, } from '../../../../Shared/index'

jest.mock('string-strip-html', () => ({
  default: jest.fn()
}))

const mockRuleSet = {
  id: 1,
  name: 'remove all instances of "it contains methane"',
  feedback: 'Revise your work. Delete the phrase "it contains methane" because it repeats the first part of the sentence',
  rules: [
    { id: 1, regex_text: 'it contain(s)? methane gas', case_sensitive: false },
    { id: 2, regex_text: 'another reg(ex) line', case_sensitive: true },
    { id: 3, regex_text: 'some m?ore reg(ex', case_sensitive: false }
  ],
  prompts: [
    { id: 1, conjunction: 'because' },
    { id:2, conjunction: 'but' }
  ]
}
const mockProps = {
  activityRuleSet: mockRuleSet,
  activityData: {
    title: 'test',
    scored_level: '7',
    target_level: 7
  },
  closeModal: jest.fn(),
  submitRuleSet: jest.fn()
};

describe('RuleSetForm component', () => {
  const container = shallow(<RuleSetForm {...mockProps} />);

  it('should render RuleSetForm', () => {
    expect(container).toMatchSnapshot();
  });

  it('should render an Input, TextEditor or checkbox component for each field', () => {
    // Input: Name (1)
    // TextEditor: Feedback (1)
    // RegexSection (1)
    expect(container.find(Input).length).toEqual(1);
    expect(container.find(TextEditor).length).toEqual(1);
    expect(container.find(RegexSection).length).toEqual(1);
  });
  it('clicking the "x" button or "close" button should call closeModal prop', () => {
    container.find('#activity-close-button').simulate('click');
    container.find('#activity-cancel-button').simulate('click');
    expect(mockProps.closeModal).toHaveBeenCalledTimes(2);
  });
  it('clicking submit button should submit activity', () => {
    container.find('#activity-submit-button').simulate('click');
    expect(mockProps.closeModal).toHaveBeenCalled();
  });
});
