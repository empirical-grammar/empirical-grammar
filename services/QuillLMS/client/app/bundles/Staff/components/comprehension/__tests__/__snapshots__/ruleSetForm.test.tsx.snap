// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RuleSetForm component should render RuleSetForm 1`] = `
<div
  className="ruleset-form-container"
>
  <div
    className="close-button-container"
  >
    <button
      className="quill-button fun primary contained"
      id="activity-close-button"
      onClick={[MockFunction]}
      type="submit"
    >
      x
    </button>
  </div>
  <form
    className="ruleset-form"
  >
    <Input
      autoComplete="on"
      className="name-input"
      handleChange={[Function]}
      label="Name"
      value="remove all instances of \\"it contains methane\\""
    />
    <p
      className="form-subsection-label"
    >
      Feedback
    </p>
    <TextEditor
      ContentState={[Function]}
      EditorState={[Function]}
      handleTextChange={[Function]}
      key="feedback-description"
      text="Revise your work. Delete the phrase \\"it contains methane\\" because it repeats the first part of the sentence"
    />
    <p
      className="form-subsection-label"
    >
      Apply To Stems
    </p>
    <div
      className="checkboxes-container"
    >
      <div
        className="checkbox-container"
      >
        <label
          id="stem-label-1"
        >
          Because
        </label>
        <input
          aria-labelledby="stem-label-1"
          name="Because"
          onChange={[Function]}
          type="checkbox"
          value="because"
        />
      </div>
      <div
        className="checkbox-container"
      >
        <label
          id="stem-label-2"
        >
          But
        </label>
        <input
          aria-labelledby="stem-label-2"
          name="But"
          onChange={[Function]}
          type="checkbox"
          value="but"
        />
      </div>
      <div
        className="checkbox-container"
      >
        <label
          id="stem-label-3"
        >
          So
        </label>
        <input
          aria-labelledby="stem-label-3"
          name="So"
          onChange={[Function]}
          type="checkbox"
          value="so"
        />
      </div>
    </div>
    <p
      className="form-subsection-label"
      id="regex-rules-label"
    >
      Regex Rules
    </p>
    <RegexSection
      errors={Object {}}
      handleAddRegexInput={[Function]}
      handleDeleteRegexRule={[Function]}
      handleSetRegexRule={[Function]}
      regexRules={Object {}}
    />
    <div
      className="submit-button-container"
    >
      <button
        className="quill-button fun primary contained"
        id="activity-submit-button"
        onClick={[Function]}
        type="button"
      >
        Submit
      </button>
      <button
        className="quill-button fun primary contained"
        id="activity-cancel-button"
        onClick={[MockFunction]}
        type="submit"
      >
        Cancel
      </button>
    </div>
  </form>
</div>
`;
